posthog/api/exports.py

```diff
@@ -0,0 +1,125 @@
+from datetime import datetime, timedelta
+import json
+from typing import Any, Dict
+
+import celery
+from django.core.serializers.json import DjangoJSONEncoder
+from django.http import HttpResponse
+from rest_framework import mixins, serializers, viewsets
+from rest_framework.authentication import BasicAuthentication, SessionAuthentication
+from rest_framework.decorators import action
+from rest_framework.exceptions import ValidationError
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.renderers import TemplateHTMLRenderer
+from rest_framework.request import Request
+from rest_framework.response import Response
+from posthog import settings
+
+from posthog.api.dashboard import DashboardSerializer
+from posthog.api.insight import InsightSerializer
+from posthog.api.routing import StructuredViewSetMixin
+from posthog.auth import PersonalAPIKeyAuthentication
+from posthog.event_usage import report_user_action
+from posthog.models.exported_asset import ExportedAsset
+from posthog.permissions import ProjectMembershipNecessaryPermissions, TeamMemberAccessPermission
+from posthog.tasks import exporter
+from posthog.utils import render_template
+
+
+class ExportedAssetSerializer(serializers.ModelSerializer):
+    """Standard ExportedAsset serializer that doesn't return content."""
+
+    class Meta:
+        model = ExportedAsset
+        fields = ["id", "dashboard", "insight", "export_format", "created_at", "has_content"]
+        read_only_fields = ["id", "created_at", "has_content"]
+
+    def validate(self, attrs):
+        if not attrs.get("dashboard") and not attrs.get("insight"):
+            raise ValidationError("Either dashboard or insight is required for an export")
+
+        return attrs
```

Comment:
Is it possible to use this API create an export for an insight that has a different team_id than the export?

Reply:
Ooh I don't know. I mostly referenced the other API endpoints when building this. Is there something I can refer to elsewhere? How are the other endpoints ensuring this?